srcutil/gen_command_info.py

```diff
@@ -0,0 +1,126 @@
+        header.write('#pragma once\n')
+        header.write('#include "redismodule.h"\n\n')
+        for name in command_names:
+            header.write(f'{get_function_signature(name)};\n')
+
+def generate_history(file, changes):
+    with Scope('.history = (RedisModuleCommandHistoryEntry[])', ',', file) as history:
+        for change in changes:
+            version, what = change
+            history.write(f'{{"{version}", "{what}"}},\n')
+        history.write('{0}\n')
+
+def generate_arguments(file, member, arguments):
+    min_arity = 0
+    with Scope(f'.{member} = (RedisModuleCommandArg[])', ',', file) as args_scope:
+        for arg in arguments:
+            with Scope('', ',', file) as arg_scope:
+                for string_arg in ['name', 'token', 'summary', 'since', 'deprecated_since', 'display_text']:
+                    if string_arg in arg:
+                        arg_scope.write(f'.{string_arg} = "{arg[string_arg]}",\n')
+                if 'type' in arg:
+                    type_text = arg['type'].replace('-', '_').upper()
+                    arg_scope.write(f'.type = REDISMODULE_ARG_TYPE_{type_text},\n')
+                flags = []
+                if 'optional' not in arg:
+                    min_arity += 1
+                for flag in ['optional', 'multiple', 'multiple-token']:
+                    if flag in arg:
+                        flag_text = flag.replace('-', '_').upper()
+                        flags.append(f'REDISMODULE_CMD_ARG_{flag_text}')
+                if len(flags) > 0:
+                    flags_text = ' | '.join(flags)
+                    arg_scope.write(f'.flags = {flags_text},\n')
+                if 'arguments' in arg:
+                    generate_arguments(file, 'subargs', arg['arguments'])
+        args_scope.write('{0}\n')
+    return min_arity
+
+def generate_redis_module_command_info(cmd_info, file):
+    with Scope('const RedisModuleCommandInfo info = ', ';', file) as info:
+        info.write('.version = REDISMODULE_COMMAND_INFO_VERSION,\n')
+        for key, value in cmd_info.items():
+            if key == 'group':
+                continue
+            if type(value) == str:
+                info.write(f'.{key} = "{value}",\n')
+            elif key == 'history':
+                generate_history(file, value)
+            elif key == 'arguments':
+                min_arity = generate_arguments(file, 'args', value)
+                info.write(f'.arity = -{min_arity},\n')
+
+def generate_command_info_definition(name, info, file):
+    signature = get_function_signature(name)
+    file.write(f'// Info for {name}\n')
+    with Scope(signature + ' ', '\n', file) as function:
+        generate_redis_module_command_info(info, file)
+        # If RedisModule_SetCommandInfo is not available, quietly return REDISMODULE_OK
+        with Scope('if (RedisModule_SetCommandInfo == NULL)', '', file) as no_set_command_info:
+            no_set_command_info.write('return REDISMODULE_OK;\n')
```

Comment:
Basically we should be running with Redis versions that have this RedisModuleAPI so this check could be removed